ngraph/core/include/ngraph/op/constant.hpp

```
@@ -352,13 +404,112 @@ namespace ngraph
-                          typename DataStorageType = fundamental_type_for<Type>>
+                          typename DataStorageType = fundamental_type_for<Type>,
+                          typename std::enable_if<Type != element::Type_t::u1 &&
+                                                      Type != element::Type_t::u4 &&
+                                                      Type != element::Type_t::i4,
+                                                  bool>::type = true>
                 void fill_data(const T& value)
                 {
                     const auto size = shape_size(m_shape);
-                    std::fill_n(get_data_ptr_nc<Type>(), size, static_cast<DataStorageType>(value));
+                    const auto v = static_cast<DataStorageType>(value);
+                    std::fill_n(get_data_ptr_nc<Type>(), size, v);
+                }
+
+                template <element::Type_t Type,
+                          typename T,
+                          typename DataStorageType = fundamental_type_for<Type>,
+                          typename std::enable_if<Type == element::Type_t::u1, bool>::type = true>
+                void fill_data(const T& value)
+                {
+                    const auto size = (shape_size(m_shape) + 7) / 8;
+                    const DataStorageType v = value ? 0xFF : 0x00;
+                    std::fill_n(get_data_ptr_nc<Type>(), size, v);
+                }
+
+                template <element::Type_t Type,
+                          typename T,
+                          typename DataStorageType = fundamental_type_for<Type>,
+                          typename std::enable_if<Type == element::Type_t::u4 ||
+                                                      Type == element::Type_t::i4,
+                                                  bool>::type = true>
+                void fill_data(const T& value)
+                {
+                    const auto size = (shape_size(m_shape) + 1) / 2;
```

I think we need to throw an exception if original value is out of range for LP types.

---

OK